python/tidy/linting_report.py

```diff
@@ -42,42 +42,38 @@ def emit_annotation(
         file_name: str,
         line_start: int,
         line_end: int | None = None,
-        annotation_level: str = "error",
+        annotation_level: Literal["notice", "warning", "error"] = "error",
         column_start: int | None = None,
         column_end: int | None = None,
-    ):
+    ) -> None:
         if self.total_count >= self.limit:
             return
 
         if line_end is None:
             line_end = line_start
 
-        annotation: GithubAnnotation = {
-            "title": f"{self.annotation_prefix}: {self.escape(title)}",
-            "message": self.escape(message),
-            "file_name": self.clean_path(file_name),
-            "line_start": line_start,
-            "line_end": line_end,
-            "level": annotation_level,
-        }
-
-        if line_start == line_end and column_start is not None and column_end is not None:
-            annotation["column_start"] = column_start
-            annotation["column_end"] = column_end
+        annotation = GithubAnnotation(
+            title=f"{self.annotation_prefix}: {self.escape(title)}",
+            message=self.escape(message),
+            file_name=self.clean_path(file_name),
+            line_start=line_start,
+            line_end=line_end,
+            level=annotation_level,
+            column_start=column_start if line_start == line_end and column_start is not None else None,
```

Comment:
```suggestion
            column_start=column_start if line_start == line_end else None,
```

The `column_start is not None` condition is redundant, as the whole expression will evaluate to `None` anyway.

Also, where does the `line_start == line_end` constraint come from? Is that required by github annotation?